mining.md

```diff
@@ -252,45 +252,61 @@ In order to determine that the mined block was generated by an elegible miner, o
-	i. 	# Determine total storage L rounds back
-  		S <-- storageMarket(N-L)
-  		p_n <-- S.GetTotalStorage()
-  ii. # Determine miner's power L rounds back
-  		miner <-- block.miner
-  		p_m <-- S.PowerLookup(miner)
-  iii. # Get power fraction
-  		p_f <-- p_m/p_n
-2. Get the appropriate ticket from the ticket chain
+1. Ensure the miner was not slashed since L.
+		i. # Fetch the latest state and the miner
+			S <-- storageMarket(N)
+			M <-- storageMinerActor(PK)
+		ii. # Check for a reported fault
+			if !S.IsMiner(PK)
+				return 0
+		iii. # Check for an unreported storage fault
+			if M.ProvingPeriodEnd < N
+				return 0
+2. Determine the miner's power fraction
+		i. 	# Determine total storage this round
+  			S_n <-- storageMarket(N)
+	  		p_n <-- S_n.GetTotalStorage()
+		ii. # Determine miner's power at last valid PoSt submission
+  			miner <-- block.miner
+  			S_m <-- storageMarket(miner.provingPeriodEnd - provingPeriodDuration(miner.SectorSize)
+	  		p_m <-- S_m.PowerLookup(miner)
+		iii. # Get power fraction
+  			p_f <-- p_m/p_n
+3. Get the appropriate ticket from the ticket chain
 		i. 	# Get the tipset K rounds back
 				appropriateTipset <-- lookback(K)
 		ii. # Take its min ticket (already validated)
 				scratchedTicket <-- appropriateTipset.minTicket()
-3. Ensure that the scratched ticket is a winner
+4. Ensure that the scratched ticket is a winner
 		i.	# get the deterministic output from the election proof
-				VRFOutput <-- ECVRF_proof_to_hash(electionProof.VRFProof)
+			VRFOutput <-- ECVRF_proof_to_hash(electionProof.VRFProof)
 		ii. # map the VRFOutput onto [0,1]
   			scratchValue <-- VRFOutput / {1}^HashLen
-  	iii. # Compare the miner's scratchValue to the miner's power fraction
+	  	iii. # Compare the miner's scratchValue to the miner's power fraction
   			if scratchValue > p_f:
     			return 0
-4. If winning ticket, verify Election Proof validity
+5. If winning ticket, verify Election Proof validity
 		i. 	# generate the VRFInput from the scratched ticket
-				input <-- scratchedTicket.VDFOutput | "e"
+			input <-- scratchedTicket.VDFOutput | "e"
 		ii. # Check that the election proof was correctly generated by the miner
     		# using the appropriate ticket
     		VRFState <-- ECVRF_Verify(miner.PK, electionProof.VRFProof, input)
-				if VRFState == "INVALID":
-					return 0
+			if VRFState == "INVALID":
+				return 0
 5. Everything checks out, it's a valid election proof
 		return 1
 ```
 
+Note: For clarity we explicit what may happen when a miner fails to submit a PoSt:
+- If they are currently late to submit a PoSt, 1.iii above is triggered and they will not mine.
+- If they are late to submit a PoSt and are slashed/reported for it, then they will be taken out of the power table (at least temporarily) and 1.ii will be triggered.
+- If they are late to submit a PoSt but the fault was unreported before they finally submitted it, then they can resume mining normally (they got lucky).
```

Comment:
Power should be lost for any period, even if they are not slashed explicitly. I am not sure how you end up in this situation.